### PR TITLE
Properly document prototype chain of generators

### PR DESCRIPTION
**Pull request by Josh-Cena**
Add two diagrams and properly explain the prototype chain instead of handwaving it in a single paragraph.

I have done this PR a long time ago but never submitted it because https://github.com/orgs/mdn/discussions/358. This now seems to be the preferred approach?

### CODE REVIEW

**Comment by Josh-Cena** [comment]
`yarn filecheck` seems to break the SVG; see the diff in https://github.com/mdn/content/pull/34267/commits/ad5d4807de8f4ddbe0e11242dac4c221b00a2906.

**Comment by bsmth** [comment]
Having a look at the diagrams now 👀

**Comment by Josh-Cena** [comment]
By the way the `GeneratorFunction/prototype` page is unpreviewable because the preview server uses the redirects file from main 😅

**Review by bsmth — CHANGES_REQUESTED** [review]
Can we move the .mmd source and generated SVG into the shared assets repo like this example PR: https://github.com/mdn/shared-assets/pull/12/files

We can keep the SVG checked in on this branch for now.

File: files/en-us/web/javascript/reference/global_objects/generatorfunction/prototype/diagram.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...ss="outer title-state"/><line y2="-20.75" y1="-20.75" x2="108.703125" x1="-108.703125" class="divider"/><line y2="40.25" y1="40.25" x2="108.703125" x1="-108.703125" class="divider"/><g class="label"><foreignObject height="0" width="0"><div style="display: inline-block; white-space: nowrap;" xmlns="http://www.w3.org/1999/xhtml"><span class="nodeLabel"></span></div></foreignObject><foreignObject transform="translate( -70.48046875, -43.75)" height="18.5" width="140.9609375" class="classTitle"><div style="display: inline-block; white-space: nowrap;" xmlns="http://www.w3.org/1999/xhtml"><span class="nodeLabel">GeneratorFunction</span></div></foreignObject><foreignObject transform="translate( -101.203125, -9.25)" height="18.5" width="71.7265625"><div style="display: inline-block; white-space: nowrap;" xmlns="http://www.w3.org/1999/xhtml"><span class="nodeLabel">length = 1</span></div></foreignObject><foreignObject transform="translate( -101.203125, 13.25)" height="18.5" width="202.40625"><div style="display: inline-block; white-space: nowrap;" xmlns="http://www.w3.org/1999/xhtml"><span class="nodeLabel">name = "GeneratorFunction"</span></div></foreignObject></g></g></a></g></g></g></svg>
```
Comment (bsmth) [inline files/en-us/web/javascript/reference/global_objects/generatorfunction/prototype/diagram.svg:1]: Same as above

```suggestion
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" id="my-svg" width="100%" aria-roledescription="classDiagram" style="max-width:766.57px;background-color:#fff" viewBox="0 0 766.57 1024"><defs><path id="reuse-0" d="m18 7-9 6-8-6 8-6Z"/></defs><style>#my-svg{fill:#000}#my-svg .edge-pattern-solid{stroke-dasharray:0}#my-svg .edge-pattern-dashed{stroke-dasharray:3}#my-svg .marker{fill:#000;stroke:#000}#my-svg,#my-svg svg{font-family:&quot;trebuchet ms&quot;,verdana,arial,sans-serif;font-size:16px}#my-svg .edgeLabel,#my-svg .nodeLabel{color:#000}#my-svg .edgeLabel .label span{background:#fff}#my-svg .classTitle{font-weight:bolder}#my-svg .node circle,#my-svg .node path{fill:#fff;stroke:#000;stroke-width:1px}#my-svg .divider,#my-svg .relation{stroke:#000;stroke-width:1}#my-svg g.clickable{cursor:pointer}#my-svg .relation{fill:none}#my-svg .composition,#my-svg .dependency{fill:#000!important;stroke:#000!important;stroke-width:1}#my-svg .aggregation,#my-svg .extension,#my-svg .lollipop{fill:transparent!important;stroke:#000!important;stroke-width:1}#my-svg .lollipop{fill:#fff!important}#my-svg .edgeTerminals{font-size:11px;line-height:initial}#my-svg :root{--mermaid-font-family:&quot;trebuchet ms&quot;,verdana,arial,sans-serif}</style><defs><marker id="my-svg_classDiagram-aggregationStart" class="marker aggregation classDiagram" markerHeight="240" markerWidth="190" orient="auto" refX="18" refY="7"><use xlink:href="#reuse-0"/></marker></defs><defs><marker id="my-svg_classDiagram-aggregationEnd" class="marker aggregation classDiagram" markerHeight="28" markerWidth="20" orient="auto" refX="1" refY="7"><use xlink:href="#reuse-0"/></marker></defs><defs><marker id="my-svg_classDiagram-extensionStart" class="marker extension classDiagram" markerHeight="240" markerWidth="190" orient="auto" refX="18" refY="7"><path d="m1 7 17 6V1Z"/></marker></defs><defs><marker id="my-svg_classDiagram-extensionEnd" class="marker extension classDiagram" markerHeight="28" markerWidth="20" orient="auto" refX="1" refY="7"><path d="M1 1v12l17-6Z"/></marker></defs><defs><marker id="my-svg_classDiagram-compositionStart" class="marker composition classDiagram" markerHeight="240" markerWidth="190" orient="auto" refX="18" refY="7"><use xlink:href="#reuse-0"/></marker></defs><defs><marker id="my-svg_classDiagram-compositionEnd" class="marker composition classDiagram" markerHeight="28" markerWidth="20" orient="auto" refX="1" refY="7"><use xlink:href="#reuse-0"/></marker></defs><defs><marker id="my-svg_classDiagram-dependencyStart" class="marker dependency classDiagram" markerHeight="240" markerWidth="190" orient="auto" refX="6" refY="7"><path d="m5 7 4 6-8-6 8-6Z"/></marker></defs><defs><marker id="my-svg_classDiagram-dependencyEnd" class="marker dependency classDiagram" markerHeight="28" markerWidth="20" orient="auto" refX="13" refY="7"><path d="m18 7-9 6 5-6-5-6Z"/></marker></defs><defs><marker id="my-svg_classDiagram-lollipopStart" class="marker lollipop classDiagram" markerHeight="240" markerWidth="190" orient="auto" refX="13" refY="7"><circle cx="7" cy="7" r="6" fill="transparent"/></marker></defs><defs><marker id="my-svg_classDiagram-lollipopEnd" class="marker lollipop classDiagram" markerHeight="240" markerWidth="190" orient="auto" refX="1" refY="7"><circle cx="7" cy="7" r="6" fill="transparent"/></marker></defs><g class="root"><g class="edgePaths"><path id="id_Object.prototype_Iterator.prototype_1" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="m335.417 190.323-15.127 10.279c-15.127 10.28-45.38 30.839-60.507 45.285-15.127 14.446-15.127 22.78-15.127 26.946V277" class="edge-pattern-solid relation"/><path id="id_Iterator.prototype_GeneratorFunction.prototype.prototype_2" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="M244.656 376v32" class="edge-pattern-solid relation"/><path id="id_GeneratorFunction.prototype.prototype_genFunc.prototype_3" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="m107.344 591.377-3.454 2.437c-3.455 2.437-10.364 7.312-13.818 17.749C86.617 622 86.617 638 86.617 646v8" class="edge-pattern-solid relation"/><path id="id_genFunc.prototype_genFunc()_4" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="M86.617 730v108.583c0 27.25 0 56.084 1.191 76.25C88.999 935 91.381 946.5 92.572 952.25l1.191 5.75" class="edge-pattern-solid relation"/><path id="id_Object.prototype_Function.prototype_5" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="m549.268 197.269 12.2 9.122c12.2 9.122 36.6 27.365 48.801 47.404 12.2 20.038 12.2 41.872 12.2 63.705v102" class="edge-pattern-solid relation"/><path id="id_Function.prototype_GeneratorFunction.prototype_6" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="m528.509 572.169-6.821 5.638c-6.821 5.639-20.464 16.916-36.361 28.638-15.898 11.722-34.051 23.888-43.128 29.972l-9.076 6.083" class="edge-pattern-solid relation"/><path id="id_GeneratorFunction.prototype_genFunc_7" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="m314.645 735.584-4.492 4.07c-4.493 4.069-13.477 12.207-22.028 20.443-8.551 8.236-16.668 16.57-20.727 20.736L263.339 785" class="edge-pattern-solid relation"/><path id="id_Function.prototype_Function_8" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="m532.274 582.036-4.115 3.994c-4.116 3.994-12.346 11.982-6.033 21.929 6.314 9.948 27.172 21.855 37.601 27.808l10.429 5.953" class="edge-pattern-solid relation"/><path id="id_Function_GeneratorFunction_9" fill="none" marker-start="url(#my-svg_classDiagram-extensionStart)" d="M642.469 753v1.167c0 1.166 0 3.5-5.682 8.833s-17.046 13.667-22.728 17.833L608.377 785" class="edge-pattern-solid relation"/><path id="id_genFunc_genFunc()_10" fill="none" marker-start="url(#my-svg_classDiagram-dependencyStart)" d="M212.688 895v4.75c0 4.75 0 14.25-11.396 25.408-11.395 11.158-34.185 23.975-45.58 30.383l-11.396 6.408" class="edge-pattern-dashed relation"/><path id="id_genFunc.prototype_genFunc_11" fill="none" marker-start="url(#my-svg_classDiagram-dependencyStart)" d="m118.253 716.353 6.9 7.275c6.9 7.274 20.7 21.823 30.469 33.264 9.77 11.441 15.51 19.775 18.38 23.941l2.87 4.167" class="edge-pattern-solid relation"/><path id="id_GeneratorFunction.prototype.prototype_GeneratorFunction.prototype_12" fill="none" marker-start="url(#my-svg_classDiagram-dependencyStart)" d="m236.362 586.976-.284 3.171c-.284 3.17-.853 9.512 9.768 18.766 10.622 9.254 32.433 21.42 43.339 27.504l10.905 6.083" class="edge-pattern-solid relation"/><path id="id_GeneratorFunction.prototype.prototype_GeneratorFunction.prototype_13" fill="none" marker-end="url(#my-svg_classDiagram-dependencyEnd)" d="m336.229 581 4.411 4.167c4.411 4.166 13.233 12.5 18.306 21.758 5.072 9.258 6.394 19.442 7.056 24.533l.661 5.092" class="edge-pattern-solid relation"/><path id="id_GeneratorFunction.prototype_GeneratorFunction_14" fill="none" marker-start="url(#my-svg_classDiagram-dependencyStart)" d="m402.195 728.536 3.397 5.244c3.397 5.244 10.192 15.732 19.806 25.143 9.614 9.41 22.047 17.744 28.264 21.91l6.217 4.167" class="edge-pattern-solid relation"/><path id="id_GeneratorFunction.prototype_GeneratorFunction_15" fill="none" marker-end="url(#my-svg_classDiagram-dependencyEnd)" d="m443.642 723.5 10.657 6.083c10.656 6.084 31.97 18.25 43.882 27.568 11.912 9.318 14.422 15.787 15.677 19.021l1.255 3.234" class="edge-pattern-solid relation"/><path id="id_Function.prototype_Function_16" fill="none" marker-start="url(#my-svg_classDiagram-dependencyStart)" d="m629.731 575.476.456 5.087c.457 5.088 1.369 15.262 2.367 24.516.997 9.254 2.079 17.588 2.62 21.754l.542 4.167" class="edge-pattern-solid relation"/><path id="id_Function.prototype_Function_17" fill="none" marker-end="url(#my-svg_classDiagram-dependencyEnd)" d="m642.648 569.5 1.637 6.083c1.637 6.084 4.91 18.25 6.135 27.509 1.224 9.258.399 15.608-.013 18.783l-.412 3.175" class="edge-pattern-solid relation"/></g><g class="edgeLabels"><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeLabel"><foreignObject width="74.422" height="19" class="label" transform="translate(175.477 914)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"><span class="edgeLabel">instanceof</span></span></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeTerminals"><foreignObject class="inner" style="width:81px;height:12px" transform="translate(115.284 735.02)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel">prototype</span></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeTerminals"><foreignObject class="inner" style="width:81px;height:12px" transform="translate(225.434 599.06)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel">prototype</span></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><foreignObject class="edgeTerminals" style="width:99px;height:12px" transform="translate(375.057 618.214)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel">constructor</span></div></foreignObject><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeTerminals"><foreignObject class="inner" style="width:81px;height:12px" transform="translate(395.858 746.343)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel">prototype</span></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><foreignObject class="edgeTerminals" style="width:99px;height:12px" transform="translate(516.864 757.96)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel">constructor</span></div></foreignObject><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><g class="edgeTerminals"><foreignObject class="inner" style="width:81px;height:12px" transform="translate(615.818 588.27)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel">prototype</span></div></foreignObject></g><g class="edgeLabel"><foreignObject width="0" height="0" class="label"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel"/></div></foreignObject></g><foreignObject class="edgeTerminals" style="width:99px;height:12px" transform="translate(659.57 609.08)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="edgeLabel">constructor</span></div></foreignObject></g><g class="nodes"><a xlink:href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object" data-id="Object.prototype" data-node="true" transform="translate(442.578 117.5)"><g id="classId-Object.prototype-0" class="node default clickable"><path d="M-92.273-109.5H92.274v219H-92.273z" class="outer title-state"/><path d="M-92.273-78.5H92.273m-184.546 39H92.273" class="divider"/><g class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="130.031" height="19" class="classTitle" transform="translate(-65.016 -102)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">Object.prototype</span></div></foreignObject><foreignObject width="72.219" height="19" transform="translate(-84.773 -67)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">__proto__</span></div></foreignObject><foreignObject width="127.875" height="19" transform="translate(-84.773 -32)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">hasOwnProperty()</span></div></foreignObject><foreignObject width="109.266" height="19" transform="translate(-84.773 -9)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">isPrototypeOf()</span></div></foreignObject><foreignObject width="169.547" height="19" transform="translate(-84.773 14)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">propertyIsEnumerable()</span></div></foreignObject><foreignObject width="114.734" height="19" transform="translate(-84.773 37)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">toLocaleString()</span></div></foreignObject><foreignObject width="68.281" height="19" transform="translate(-84.773 60)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">toString()</span></div></foreignObject><foreignObject width="66.875" height="19" transform="translate(-84.773 83)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">valueOf()</span></div></foreignObject></g></g></a><a xlink:href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator" data-id="Iterator.prototype" data-node="true" transform="translate(244.656 317.5)"><g id="classId-Iterator.prototype-1" class="node default clickable"><path d="M-75.039-40.5H75.039v81H-75.039z" class="outer title-state"/><path d="M-75.039-9.5H75.039m-150.078 16H75.039" class="divider"/><g class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="135.078" height="19" class="classTitle" transform="translate(-67.54 -33)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">Iterator.prototype</span></div></foreignObject><foreignObject width="109.25" height="19" transform="translate(-67.54 14)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">\[@@iterator]()</span></div></foreignObject></g></g></a><a xlink:href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator" data-id="GeneratorFunction.prototype.prototype" data-node="true" transform="translate(244.656 494.5)"><g id="classId-GeneratorFunction.prototype.prototype-2" class="node default clickable"><path d="M-157.867-86.5h315.734v173h-315.734z" class="outer title-state"/><path d="M-157.867-32.5h315.734m-315.734 39h315.734" class="divider"/><g class="label"><foreignObject width="163.844" height="19" transform="translate(-81.922 -79)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">«Generator.prototype»</span></div></foreignObject><foreignObject width="300.734" height="19" class="classTitle" transform="translate(-150.367 -56)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">GeneratorFunction.prototype.prototype</span></div></foreignObject><foreignObject width="217.875" height="19" transform="translate(-150.367 -21)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">[@@toStringTag] = &quot;Generator&quot;</span></div></foreignObject><foreignObject width="43.578" height="19" transform="translate(-150.367 14)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">next()</span></div></foreignObject><foreignObject width="56.75" height="19" transform="translate(-150.367 37)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">return()</span></div></foreignObject><foreignObject width="53.547" height="19" transform="translate(-150.367 60)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">throw()</span></div></foreignObject></g></g></a><g id="classId-genFunc.prototype-3" class="node default" data-id="genFunc.prototype" data-node="true"><path d="M8 654h157.234v58H8z" class="outer title-state"/><path d="M8 685h157.234M8 701h157.234" class="divider"/><g class="label" transform="translate(86.617 683)"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="142.234" height="19" class="classTitle" transform="translate(-71.117 -21.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">genFunc.prototype</span></div></foreignObject></g></g><g id="classId-genFunc()-4" class="node default" data-id="genFunc()" data-node="true"><path d="M55.223 958h89.094v58H55.223z" class="outer title-state"/><path d="M55.223 989h89.094m-89.094 16h89.094" class="divider"/><g class="label" transform="translate(99.77 987)"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="74.094" height="19" class="classTitle" transform="translate(-37.047 -21.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">genFunc()</span></div></foreignObject></g></g><a xlink:href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function" data-id="Function.prototype" data-node="true" transform="translate(622.469 494.5)"><g id="classId-Function.prototype-5" class="node default clickable"><path d="M-80.086-75H80.086V75H-80.086z" class="outer title-state"/><path d="M-80.086-44H80.086M-80.086-28H80.086" class="divider"/><g class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="145.172" height="19" class="classTitle" transform="translate(-72.586 -67.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">Function.prototype</span></div></foreignObject><foreignObject width="137.281" height="19" transform="translate(-72.586 -20.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">\[@@hasInstance]()</span></div></foreignObject><foreignObject width="50.594" height="19" transform="translate(-72.586 2.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">apply()</span></div></foreignObject><foreignObject width="42.891" height="19" transform="translate(-72.586 25.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">bind()</span></div></foreignObject><foreignObject width="37.516" height="19" transform="translate(-72.586 48.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">call()</span></div></foreignObject></g></g></a><a xlink:href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/GeneratorFunction" data-id="GeneratorFunction.prototype" data-node="true" transform="translate(372.695 683)"><g id="classId-GeneratorFunction.prototype-6" class="node default clickable"><path d="M-147.461-40.5h294.922v81h-294.922z" class="outer title-state"/><path d="M-147.461-9.5h294.922m-294.922 39h294.922" class="divider"/><g class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="220.844" height="19" class="classTitle" transform="translate(-110.422 -33)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">GeneratorFunction.prototype</span></div></foreignObject><foreignObject width="279.922" height="19" transform="translate(-139.96 2)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">[@@toStringTag] = &quot;GeneratorFunction&quot;</span></div></foreignObject></g></g></a><g id="classId-genFunc-7" class="node default" data-id="genFunc" data-node="true"><path d="M121.618 785h182.141v104H121.618z" class="outer title-state"/><path d="M121.618 816h182.14m-182.14 62h182.14" class="divider"/><g class="label" transform="translate(212.688 837)"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="167.141" height="19" class="classTitle" transform="translate(-83.57 -44.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">function* genFunc() {}</span></div></foreignObject><foreignObject width="71.734" height="19" transform="translate(-83.57 -9.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">length = 0</span></div></foreignObject><foreignObject width="126.891" height="19" transform="translate(-83.57 13.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">name = &quot;genFunc&quot;</span></div></foreignObject></g></g><a xlink:href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function" data-id="Function" data-node="true" transform="translate(642.469 683)"><g id="classId-Function-8" class="node default clickable"><path d="M-72.313-52H72.312V52H-72.313z" class="outer title-state"/><path d="M-72.313-21H72.313M-72.313 41H72.313" class="divider"/><g class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="65.297" height="19" class="classTitle" transform="translate(-32.648 -44.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">Function</span></div></foreignObject><foreignObject width="71.734" height="19" transform="translate(-64.813 -9.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">length = 1</span></div></foreignObject><foreignObject width="129.625" height="19" transform="translate(-64.813 13.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">name = &quot;Function&quot;</span></div></foreignObject></g></g></a><a xlink:href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/GeneratorFunction" data-id="GeneratorFunction" data-node="true" transform="translate(537.465 837)"><g id="classId-GeneratorFunction-9" class="node default clickable"><path d="M-108.703-52h217.406V52h-217.406z" class="outer title-state"/><path d="M-108.703-21h217.406m-217.406 62h217.406" class="divider"/><g class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel"/></div></foreignObject><foreignObject width="140.969" height="19" class="classTitle" transform="translate(-70.484 -44.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">GeneratorFunction</span></div></foreignObject><foreignObject width="71.734" height="19" transform="translate(-101.203 -9.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">length = 1</span></div></foreignObject><foreignObject width="202.406" height="19" transform="translate(-101.203 13.5)"><div xmlns="http://www.w3.org/1999/xhtml" display="inline-block" style="white-space:nowrap"><span class="nodeLabel">name = &quot;GeneratorFunction&quot;</span></div></foreignObject></g></g></a></g></g></svg>
```

**Comment by Josh-Cena** [comment]
=> https://github.com/mdn/shared-assets/pull/15

**Comment by bsmth** [comment]
These are now available here if you want to include them as external images:

* https://mdn.github.io/shared-assets/images/diagrams/javascript/generatorfunction/prototype-chain.svg
* https://mdn.github.io/shared-assets/images/diagrams/javascript/asyncgeneratorfunction/prototype-chain.svg

Alternatively, I've added minified versions above that pass yarn filecheck locally.

**Comment by Josh-Cena** [comment]
How do I include the image? The shared asset doesn't seem to have the right CORS.

**Review by bsmth — APPROVED** [review]

**Comment by bsmth** [comment]
> How do I include the image? The shared asset doesn't seem to have the right CORS.

That looks fine to me on localhost, the CSP rule is not yet updated on the dev environment, so the previews are not yet fixed, but it's fine on prod. For instance: https://developer.mozilla.org/en-US/docs/Web/HTTP/Overview

**Comment by Josh-Cena** [comment]
Ah ok, sounds good then!

**Comment by bsmth** [comment]
One last thing: is it known that there are bad BCD keys:

https://pr34267.content.dev.mdn.mozit.cloud/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGeneratorFunction/prototype#browser_compatibility

**Comment by Josh-Cena** [comment]
You're right. I'm not particularly enthusiastic about adding compat data for this, so I'm just going to use `spec-urls` instead.

**Pull request merged**